src/puj/push/unit.clj

```diff
@@ -46,33 +46,38 @@
   (required-stacks [this] "A set of stack names relevant to the instruction."))
 
 
-; @TODO: Allow for non-collection instruction results.
-
+; 
 (defrecord SimpleInstruction [input-stacks output-stacks opens func]
```

Comment:
Can we do the same thing for `input-stacks`?

Reply:
I thought about this, and decided against it, since leaving it as-is makes the inputs look like the arguments list to a function, which seems natural even if there's only one argument. That said, it would be an easy change if we want this functionality.

Reply:
Good point. Let's leave it.

Reply:
Also, the main motivation for me for changing the outputs was that the function supplied to the instruction gets much simpler -- for example, you can just use the + function instead of an anonymous function that wraps that result in a vector. See https://github.com/push-language/Puj/pull/3/files#diff-5c9e838d1f70538342354ae87e90396fL13

I don't think there's an equivalent benefit for `input-stacks`.

Reply:
I love that! In #5 I added coercion functions to each type so that we can go even further. For example `/` may produce non-int values when given int arguments, but if we know how to "coerce" to int if the `output-stacks` is `:int` then we can do even more reuse.